src/main/java/com/daitangroup/api/controller/PersonController.java

```diff
@@ -1,35 +1,124 @@
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RequestMethod;
-import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.RequestParam;
 import org.springframework.web.bind.annotation.RestController;
-
 import java.util.Optional;
 
 @RestController
-@RequestMapping("/person")
+@RequestMapping("/persons")
 public class PersonController {
 
     @Autowired
-    private PersonRepository repository;
+    private GarageService garageService;
 
     @RequestMapping(method = RequestMethod.GET, value = "/{id}")
-    @ResponseBody
-    public ResponseEntity<Person> find(@PathVariable Long id) {
+    public ResponseEntity<Person> findPersonById(@PathVariable Long id) {
+
+        Optional<Person> personResult = garageService.findPersonById(id);
+
+        if (personResult.isPresent()) {
+            return new ResponseEntity<>(personResult.get(), HttpStatus.OK);
+        }
+
+        return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+    }
+
+    @RequestMapping(method = RequestMethod.GET)
+    public ResponseEntity<Iterable> findAllPersons() {
+
+        Iterable<Person> personsResult = garageService.findAllPersons();
+
+        if (personsResult.iterator().hasNext()) {
+            return new ResponseEntity<>(personsResult, HttpStatus.OK);
+        }
+
+        return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+    }
+
+    @RequestMapping(method = RequestMethod.GET, value = "/cpf/{cpf}")
+    public ResponseEntity<Person> findPersonByCpf(@PathVariable String cpf) {
+
+        Optional<Person> personResult = garageService.findPersonByCpf(cpf);
+
+        if (personResult.isPresent()) {
+            return new ResponseEntity<>(personResult.get(), HttpStatus.OK);
+        }
+
+        return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+    }
+
+    @RequestMapping(method = RequestMethod.POST, value = "/create")
+    public ResponseEntity<PersonReturn> createPerson(@RequestBody Person person) {
+
+        PersonReturn personToReturn = new PersonReturn();
+
+        try {
+            Person personSaved = garageService.createPerson(person);
+            personToReturn.setPerson(personSaved);
+            return new ResponseEntity<>(personToReturn, HttpStatus.OK);
 
-        Optional<Person> person = repository.findById(id);
+        } catch (CpfAlreadyRegistredException ex) {
 
-        if (person.isPresent()) {
-            return new ResponseEntity<>(person.get(), HttpStatus.OK);
+            personToReturn.setPerson(person);
+            personToReturn.setErrorMessage(ex.getMessage());
+            return new ResponseEntity<>(personToReturn, HttpStatus.NOT_ACCEPTABLE);
```

Comment:
You could also use the [409/CONFLICT](https://httpstatuses.com/409) code. 

> The request could not be completed due to a conflict with the current state of the target resource. This code is used in situations where the user might be able to resolve the conflict and resubmit the request.
> 
> The server SHOULD generate a payload that includes enough information for a user to recognize the source of the conflict.
> 
> Conflicts are most likely to occur in response to a PUT request. For example, if versioning were being used and the representation being PUT included changes to a resource that conflict with those made by an earlier (third-party) request, the origin server might use a 409 response to indicate that it can't complete the request. In this case, the response representation would likely contain information useful for merging the differences based on the revision history.

I've seen some suggestions saying that is best to return as payload the conflicting resource (in this case, the already existing person) or a link to it, along with the error message. But I'm satisfied with the current solution

Reply:
Ok. I will use the conflict code. Makes more sense since the user can resolve the conflict.
I will take note about sending the conflicting resource along with the error message. Thanks for the tips Daniel.